app/ts/components/subcomponents/ConfigureRpcConnection.tsx

```diff
@@ -21,9 +25,55 @@ const ConfigureRpcContext = createContext<ConfigureRpcContext | undefined>(undef
 const RpcQueryProvider = ({ children }: { children: ComponentChildren }) => {
 	const { value: rpcQuery, waitFor, reset: resetRpcQuery } = useAsyncState<Network>()
 
+	const checkServerAvailability = async (url: string) => {
+		try {
+			const provider = new JsonRpcProvider(url)
+			return await provider.getNetwork()
+		} catch {
+			throw new Error('Server should be reachable')
+		}
+	}
+
+	const validateEthSimulateSupport = async (url: string) => {
+		// create temporary rpc server from an RPC URL
+		const interimRpc: RpcEntry = { httpsRpc: url, name: '', chainId: 0n, currencyName: '', currencyTicker: '', primary: false, minimized: false, }
```

Comment:
I think we could use `return ethSimulateV1Result.parse(await this.requestHandler.jsonRpcRequest(call))` directly, instead of using the EthereumClientService. `requestHandler` doesn't really need to know chain id either, so that could be refactored so that it only takes the URL, and no other fields.

Reply:
The refactor for `requestHandler` can be tedious for this PR. created an issue here
https://github.com/DarkFlorist/TheInterceptor/issues/1037

Reply:
> I think we could use `return ethSimulateV1Result.parse(await this.requestHandler.jsonRpcRequest(call))` directly, instead of using the EthereumClientService. `requestHandler` doesn't really need to know chain id either, so that could be refactored so that it only takes the URL, and no other fields.

now using requestHandler directly on [ee9533d](https://github.com/DarkFlorist/TheInterceptor/pull/1030/commits/ee9533d68ecb51098755d1e392dc9a832f179c2d)